ui/src/mocks/handlers.ts

```diff
@@ -1,27 +1,39 @@
 import { rest } from "msw";
 import { RecipesModel } from "../models";
-// interface LoginBody {
-//   username: string;
-// }
-// interface LoginResponse {
-//   username: string;
-//   firstName: string;
-// }
+
+const recipes: RecipesModel[] = [
+  {
+    title: "Spaghetti",
+    ingredients: ["tomatoes", "pasta"],
+    description: "",
+    imgUrl: "",
+  },
+  {
+    title: "Grilled chicken",
+    ingredients: ["chicken", "olive"],
+    description: "",
+    imgUrl: "",
+  },
+];
 
 export const handlers = [
-  // rest.post<LoginBody, LoginResponse>("/login", (req, res, ctx) => {
-  //   const { username } = req.body;
-  //   return res(
-  //     ctx.json({
-  //       username,
-  //       firstName: "John",
-  //     })
-  //   );
-  // }),
   rest.get<RecipesModel>("/api/products?q=mil", (req, res, ctx) => {
     return res(ctx.json([["milk"] as string[]]));
   }),
   rest.get<RecipesModel>("/api/recipes", (req, res, ctx) => {
-    return res(ctx.json([{ title: "Some recipe" } as RecipesModel]));
+    const query = req.url.searchParams.getAll("q");
+    let response = recipes;
+
+    if (query.length > 0) {
+      response = recipes.filter((recipe: RecipesModel) =>
+        recipe.ingredients
+          .flatMap((ingredient: string) =>
+            query.map((q: string) => ingredient.includes(q))
+          )
+          .reduce((acc: boolean, val: boolean) => acc || val)
```

Comment:
what is acc? we should use descriptive names

Reply:
In all other cases, I'd agree. acc is an abbreviation from accumulator. I can change it to `accumulator` if you want but in this case, I don't think that using the short name `acc` or even `a` is a bad thing:
* this is a local variable inside a lambda, so it's easy to see what represents
* `reduce()` is a built-in function, so API is well-known
* I think it's popular to use shorter names inside lambdas these days.

Reply:
Yes, I agree that lamdas should be short names, but names, not acronyms. You have to remember that something obvious to you, will not be obvious to someone else. You know API, but junior dev doesn't have to.  By using precise names you help others understand your code. Also in lampda :)

Accumolator will be ok.